src/main/java/com/yahoo/memory/CompareAndCopy.java

```diff
@@ -108,14 +108,31 @@ private static boolean equalsByBytes(final Object arr1, final long cumOff1, fina
     return true;
   }
 
+  static int hashCode(final BaseState state) {
+    state.checkValid();
+    final long lengthBytes = state.getCapacity();
+    final long offsetBytes = state.getCumulativeOffset();
+    final Object unsafeObj = state.getUnsafeObject(); //could be null
+
+    final long hash = xxHash64(unsafeObj, offsetBytes, lengthBytes);
+    return (int) hash;
+  }
+
+  static long longHashCode(final BaseState state, final long offsetBytes, final long lengthBytes) {
```

Comment:
@leventov 
As I have said before, "longHashCode" with an offset and length is a real distortion of what a "hashCode" is.  I have rename this method "xxHash64 and moved it and the above hashCode() method to the BaseState class.  This eliminates some unnecessary indirection and these new methods in the BaseState class are very simple.